### PR TITLE
Move from bootctl to mokutil when checking for Secure Boot status

### PR DESCRIPTION
**Pull request by DavidZagury**
#### What I did
Moved to use mokutil instead of bootctl as bootctl is no longer available in Bookworm.
This affected reboot scripts, and upgrade scenario.

#### How I did it
Change calls to _bootctl status_ with _mokutil --sb-state_

#### How to verify it
After fixing the scripts to check reboot:

root@sn5600:/home/admin# soft-reboot

SECURE_UPGRADE_ENABLED=1
[[ CHECK_SECURE_UPGRADE_ENABLED -ne 0 ]]
load_kernel_secure
invoke_kexec -s
packet_write_wait: port 22: Broken pipe
admin@sn5600:~$ show reboot-cause
User issued 'soft-reboot' command [User: admin, Time: Tue Jul 23 11:06:43 PM UTC 2024]

### CODE REVIEW

File: scripts/soft-reboot [hunk 1]
```diff
@@ -93,7 +93,7 @@ function clear_lingering_reboot_config()
     if [[ -f ${WARM_DIR}/${REDIS_FILE} ]]; then
         mv -f ${WARM_DIR}/${REDIS_FILE} ${WARM_DIR}/${REDIS_FILE}.${TIMESTAMP} || /bin/true
     fi
-    /sbin/kexec -u || /bin/true
```
Comment (Yarden-Z) [inline scripts/soft-reboot:96]: Why is this change required?

File: sonic_installer/bootloader/grub.py [hunk 2]
```diff
@@ -164,7 +164,8 @@ def is_secure_upgrade_image_verification_supported(self):
             if ! [ -n "$(ls -A /sys/firmware/efi/efivars 2>/dev/null)" ]; then
                 mount -t efivarfs none /sys/firmware/efi/efivars 2>/dev/null
             fi
-            SECURE_UPGRADE_ENABLED=$(bootctl status 2>/dev/null | grep -c "Secure Boot: enabled")
+            SECURE_UPGRADE_ENABLED=$(mokutil --sb-state 2>/dev/null | \
+                grep -c "enabled") || CHECK_SECURE_UPGRADE_ENABLED=$?
```
Comment (Yarden-Z) [inline sonic_installer/bootloader/grub.py:168]: From where are you getting the "CHECK_SECURE_UPGRADE_ENABLED" variable?

File: scripts/soft-reboot [hunk 3]
```diff
@@ -93,7 +93,7 @@ function clear_lingering_reboot_config()
...
+    /sbin/kexec -u -a || /bin/true
```
Comment (Yarden-Z) [inline scripts/soft-reboot:96]: Why is this change required?

File: sonic_installer/bootloader/grub.py (hunk 2 above)
Comment (DavidZagury) [reply sonic_installer/bootloader/grub.py]: @Yarden-Z  Remove as unneeded

**Comment by qiluo-msft** [comment]
Could you link this PR to HLD PR?

File: scripts/soft-reboot (hunk 3 above)
Comment (qiluo-msft) [reply scripts/soft-reboot:96]: I have the same question. If this is valid and general bug fix, we need to evaluate the backport needs. Could you separate it into a standalone PR?

File: scripts/soft-reboot [hunk 4]
```diff
@@ -147,9 +147,17 @@ function setup_reboot_variables()
...
 }
 
+function invoke_kexec() {
+    /sbin/kexec -l "$KERNEL_IMAGE" --initrd="$INITRD" --append="$BOOT_OPTIONS" $@
+}
+
 function load_kernel() {
     # Load kernel into the memory
-    /sbin/kexec -l "$KERNEL_IMAGE" --initrd="$INITRD" --append="$BOOT_OPTIONS"
+    invoke_kexec -a
```
Comment (qiluo-msft) [inline scripts/soft-reboot:156]: >-a [](http://example.com/codeflow?start=17&length=2)

why changing? If this is valid and general bug fix, we need to evaluate the backport needs. Could you separate it into a standalone PR?

File: scripts/soft-reboot (hunk 3 above)
Comment (vivekrnv) [reply scripts/soft-reboot:96]: kexec -u on systems with SB enabled is failing. Already fixed for WR here: https://github.com/sonic-net/sonic-utilities/pull/3439. 

Since we are updating soft-reboot to work on Secure boot systems. i've also added this change

File: scripts/soft-reboot (hunk 4 above)
Comment (vivekrnv) [reply scripts/soft-reboot:156]: Similar to above comment.

**Review by ycoheNvidia — APPROVED** [review]
Looks fine to me

**Review by ycoheNvidia — APPROVED** [review]

**Review by vivekrnv — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@qiluo-msft can you please review comment feedback?

**Review by qiluo-msft — APPROVED** [review]
